### ISSUE TITLE
View Layer Solo Without Effects and/or Layer Opacity

### ISSUE THREAD

**Issue by AtsusaKaneytza** [mypaint/mypaint#107]
Probably an enhancement but it'd be kind of nice if there was a way to view the contents of a layer "as is." Usually, when I'm trying to view a layer by itself, it will still display as if the layer effect is applied.
Example: If I have a layer with white paint strokes set to multiply and I press "Layer Solo" then the layer will display as invisible, instead of showing the white paint Strokes.
![1](https://cloud.githubusercontent.com/assets/8382492/4354865/14234eb4-4243-11e4-94db-14d7767d6b49.png)
![2](https://cloud.githubusercontent.com/assets/8382492/4354871/1b2e1afe-4243-11e4-83cb-763185a2af54.png)
![3](https://cloud.githubusercontent.com/assets/8382492/4354866/157dbb5a-4243-11e4-88ce-4168eefffb3c.png)

I'm suggesting this feature because sometimes I will go to a layer to select the color I used on it, but I usually have to set its opacity to 100% and remove the layer effect, select the desired color, and then reset all of the effects and opacity I had on the layer. Which is time consuming.
Quickly moving between layers and using the eyedropper on the layer preview before it displays normally again is also kind of annoying to do.

**Comment by 0ion9** [comment]
I see there has not been much discussion here. I can personally confirm the above experience.
Showing the current layer with Normal mode + 100% opacity seems like the only reliable measure to ensure you get an accurate result. There is a related issue though, which is : AFAICS, if you are eyedropping a soloed layer, you never want a mix with the background. ie. say that the layer is drawn over checkerboard pattern, and there is a 50% opacity pixel in the layer, that has a color #ff00ff. Eyedropping that pixel should get the color #ff00ff, not a 50% mix between that and the background pixel.

I guess that would require some special casing in the eyedropper to take color from the layer pixels rather than projection? I can add it as a separate issue if you want.

Regarding "save and restore sets of layer visibility flags", Photoshop has already solved this problem -- see https://helpx.adobe.com/photoshop/using/layer-comps.html.
In short:
* new comps are created with the current visibility etc states
* comps are applied via a button in the comp dialog. The previous state of the document is remembered, clicking the button a second time restores the visibility states to the remembered state.
* updating comps is done by selecting them in the comp dockable, adjusting layer visibility in the document, and hitting a button in the comp dockable to update them.
* buttons to cycle through layer comps (ie. select and apply them, in one step) are available in the comp dialog.

As someone who has never actually used PS's layer comps, it seems quite easy to understand, even reading about it for the first time at 12:50 at night.

**Comment by AtsusaKaneytza** [comment]
@achadwick  *"Should it just show the current layer (with Normal mode and 100% opacity and visible) over the checkerboard alpha backdrop, like layer blinking? That'll allow people to paint masks more easily too."*
Yes, please and thank you

@0ion9 
*"There is a related issue though, which is : AFAICS, if you are eyedropping a soloed layer, you never want a mix with the background. ie. say that the layer is drawn over checkerboard pattern, and there is a 50% opacity pixel in the layer, that has a color #ff00ff. Eyedropping that pixel should get the color #ff00ff, not a 50% mix between that and the background pixel."*
That's a very good point too. The checkerboard background functioning as "transparency" kind of defeats its purpose when the color picker acts that way.

**Comment by achadwick** [comment]
This cuts across #233. Closing in favour of that issue, where the plan is:

> [...] Taking this and #107 together, I think we should have two behaviours:
> 
> 1. When "Show Background" is ON: the layer is shown by itself, directly over the background, with only its own layer mode in effect.
> 2. When "Show Background" is OFF: the layer is shown by itself, directly over the builtin checkerboard background, with its layer mode forced to "Normal", and at 100% opacity. This is what the layer blink effect does right now.

If there are eyedropper issues, those should be reported separately because I think that's going to require a rewrite of the way the dropper tool works.
